src/devTools.js

```diff
@@ -50,10 +50,18 @@ function connectToServer(options) {
   watch();
 }
 
-export function start(options) {
+async function obtainHostname(options, hostnamePromise) {
+  if (!options.hostname) {
+    options.hostname = await hostnamePromise.catch((err) => {
+      throw new Error('Error obtaining socket hostname: ' + err.toString());
+    });
+  }
```

Comment:
First, we can reduce indentation here:

```js
if (options.hostname) {
  return;
}
options.hostname = await hostnamePromise.catch((err) => {
  throw new Error('Error obtaining socket hostname: ' + err.toString());
});
```

Second, I think you're still getting used to async functions and how to interact with them. You're mixing up promise (`.catch`) and async (`await`) syntax. We've touched on async functions several times in this review, so I'll reiterate my request from last Friday to consider reading up on async functions a bit more.

It's also not clear to me what `await hostnamePromise.catch` returns. Based on reading [the docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch) and doing some testing, it does not seem that `await promise.catch` will return the same thing as `await promise`, so I think this code is flawed. Perhaps I am reading/testing wrong... did you test this to make sure it works?

In this case, here is what I think you wanted:

```js
if (options.hostname) {
  return;
}
try {
  options.hostname = await hostnamePromise;
} catch (err) {
  throw new Error('Error obtaining socket hostname: ' + err.toString());
}
```

Reply:
I know that `.then()` and `.catch()` are usually used with `Promise` but you can also use `.catch()` with `await` like I did here. I think it looks better. **I did it on purpose** to avoid having large `try...catch` just to cover one instruction

Reply:
Is prefixing the error necessary here?

```js
try {
  options.hostname = await hostnamePromise;
} catch (err) {
  throw err;
}
```

Is the same as:
```js
options.hostname = await hostnamePromise;
```